### PR TITLE
Add baseline (shift and fill with corresponding pixels in input)

### PR DESCRIPTION
**Pull request by jeswan**

### CODE REVIEW

File: src/metrics/fid/fid_callback.py [hunk 1]
```diff
@@ -73,32 +73,39 @@ def on_validation_epoch_start(self, trainer, pl_module):
...
+                # self.to(torch.device('cpu'))
 
             #self.last_global_step = trainer.global_step
 
         else:
-            return 0
\ No newline at end of file
+            return 0
+
+    def _calc_fid(self, pl_module, *args):
```
Comment (jeswan) [inline src/metrics/fid/fid_callback.py]: @annikabrundyn This change shouldn't affect you  (slight refactor so that we can use this code with the baseline). Let me know if you see any issues

File: src/metrics/fid/fid_callback.py [hunk 2]
```diff
@@ -73,32 +73,39 @@ def on_validation_epoch_start(self, trainer, pl_module):
...
+        features = []
+
+        samples_used = 0
+        for i, (img, target) in enumerate(tqdm(self.val_dl, desc="Getting features for generated images.")):
+            samples_used += len(target)
+            if samples_used >= self.num_samples:
+                break
+
+            pred = pl_module(img.squeeze(1), *args)
```
Comment (jeswan) [inline src/metrics/fid/fid_callback.py]: @annikabrundyn Is this a suitable way to handle the 5D tensors? Does it conflict with what you are doing?

File: src/metrics/fid/fid_callback.py (hunk 1 above)
Comment (annikabrundyn) [reply src/metrics/fid/fid_callback.py]: will test this tonight to make sure before i merge :D

File: src/metrics/fid/fid_callback.py [hunk 3]
```diff
@@ -62,44 +62,45 @@ def __init__(self, pickle_dir, pickle_name, val_dl, num_samples, fid_freq):
...
-                        break
+                print(f"FID: {fid}\n")
 
-                    pred = pl_module(img.to(pl_module.device))
+                # log FID
+                pl_module.logger.experiment.add_scalar("val_fid", fid, trainer.global_step)
 
-                    feat = self.inception(pred)[0].view(pred.shape[0], -1)  # compute features
-                    features.append(feat.to('cpu'))
+    def _calc_fid(self, pl_module, *args):
```
Comment (jeswan) [inline src/metrics/fid/fid_callback.py]:
```suggestion
    def calc_validation_fid(self, pl_module, *args):
```

File: src/models/baseline/baseline_model.py [hunk 4]
```diff
@@ -83,36 +93,56 @@ def forward(self, x, x_trans, fill="copy_left"):
...
         min_x_trans = min(loss_result_dict[loss], key=loss_result_dict[loss].get)
         min_loss = loss_result_dict[loss][min_x_trans]
         final_results_dict[loss] = {"min_loss": min_loss, "min_x_trans": min_x_trans}
 
+    # Calculate metrics for the optimal shift for each loss
+    fid = FIDCallback(args.data_dir, "real_stats.pickle", val_dataloader, len(dm.val_samples), 1)
+    lpips = lpips.LPIPS(net='alex').to(device)
+
+    for loss in loss_result_dict:
+        fid_val = fid._calc_fid(model, final_results_dict[loss]["min_x_trans"])
```
Comment (jeswan) [inline src/models/baseline/baseline_model.py]:
```suggestion
        fid_val = fid.calc_validation_fid(model, final_results_dict[loss]["min_x_trans"])
```

File: src/models/baseline/baseline_model.py [hunk 5]
```diff
@@ -83,36 +93,56 @@ def forward(self, x, x_trans, fill="copy_left"):
...
             for batch_idx, sample in enumerate(val_dataloader):
                 inputs, targets = sample
-                inputs, targets = inputs.squeeze(1).to(device), targets.to(device)
+                inputs, targets = inputs.to(device), targets.to(device)
                 shifted_inputs = model(inputs, x_tran)
 
                 for loss in loss_module_dict:
                     loss_result_dict[loss][x_tran] += loss_module_dict[loss](
-                        y_true=targets.to(device), y_pred=shifted_inputs).cpu().item() / len(inputs)
+                        y_true=targets.to(device), y_pred=shifted_inputs).cpu().item()
```
Comment (jeswan) [inline src/models/baseline/baseline_model.py:114]: Check that each loss returns a mean

File: src/models/baseline/baseline_model.py [hunk 6]
```diff
@@ -83,36 +93,56 @@ def forward(self, x, x_trans, fill="copy_left"):
...
+        ssim_val, psnr_val, lpips_val = 0, 0, 0
+
+        for batch_idx, sample in tqdm(enumerate(val_dataloader), desc=f"Calculating metrics for {loss}"):
+            inputs, targets = sample
+            inputs, targets = inputs.to(device), targets.to(device)
+            shifted_inputs = model(inputs, final_results_dict[loss]["min_x_trans"])
+
+            ssim_val += ssim(shifted_inputs, targets.type(targets.dtype)).mean()
+            psnr_val += psnr(shifted_inputs, targets).mean()
+            lpips_val += lpips(shifted_inputs, targets).mean()
```
Comment (jeswan) [inline src/models/baseline/baseline_model.py:144]: TODO: Revist how metrics are aggregated

File: src/models/baseline/baseline_model.py [hunk 7]
```diff
@@ -83,36 +93,56 @@ def forward(self, x, x_trans, fill="copy_left"):
...
+
+        final_results_dict[loss]['SSIM'] = ssim_val.item() / len(val_dataloader)
+        final_results_dict[loss]['PSNR'] = psnr_val.item() / len(val_dataloader)
+        final_results_dict[loss]['LPIPS'] = lpips_val.item() / len(val_dataloader)
+        final_results_dict[loss]['FID'] = fid_val
```
Comment (jeswan) [inline src/models/baseline/baseline_model.py:151]: Check this too

**Comment by annikabrundyn** [comment]
i still need to run an example from this branch to check the fid thing! sorry will do later this week so we can merge it
